### ISSUE TITLE
Make using the cart and checkout block on same page a good experience.

### ISSUE THREAD

**Issue by jonathan-dejong** [woothemes/woocommerce#42059]
## What do you like about the Cart & Checkout blocks?
They seem to work nicely so far and out the box have a bit more updated and neater design than the original shortcodes.

## What do you think is missing from the Cart & Checkout blocks?
It's very common to want to have the cart and checkout on the same page. Having a separate cart page to the checkout page is not very advisable given that you want to give your customers as smooth checkout process as possible. 

In our trials of the blocks together on a page they play nice together (changing cart contents reflects in the checkout block automatically). But the lack of configurability, especially in the cart block, makes the experience weird. 
Here's a rough idea of what I think would be needed:

* Ability to select 1 or 2 column layout (right now it's fixed to 2). 
* Ability to remove the secondary column content of the cart block (as it is right now) with the summary and especially the "proceed to checkout" button (we are already there!!).

That's really all. 
It doesn't feel like too much work for a great improvement? 

We've got some other feedback too but to keep these on point I'll make a separate issue for that 😄

**Comment by senadir** [comment]
Thank you for the feedback.

Using Cart/Checkout in the same page might be always advisable but it should work fine. 

For hiding content, it's not possible right now, but you can use CSS to hide them, and it should work just fine.

When it comes to having the block be 1 or 2 columns, it will go from 2 to 1 when the width is smaller, you can limit the width in the editor via selecting alignment or wrapping your checkout block in a group block

<img width="319" alt="image" src="https://github.com/woocommerce/woocommerce-blocks/assets/6165348/2450f4c4-657e-4390-9c9a-15bec8d7f201">

<img width="272" alt="image" src="https://github.com/woocommerce/woocommerce-blocks/assets/6165348/acd63648-dbbf-443a-9790-591111ab706a">

I will leave this issue open for now so we can make sure we have a full answer for it in the future.

**Comment by jonathan-dejong** [comment]
Columns issue was made here instead: https://github.com/woocommerce/woocommerce-blocks/issues/11947

**Comment by jonathan-dejong** [comment]
Yes from our testing it appears they behave well when on the same page, updating its contents etc. as necessary. 

Thanks for the response on the content hiding. We are doing it with CSS for now just as you say. 
May I ask if there's any plans to officially support this approach and possibly add that capability without a CSS based solution? 

I think it's great that your team is asking for feedback but I have a hard time finding something official on what's being worked on etc. 

Last roadmap post I can find is this one which is coming in on a year old soon: https://developer.woo.com/2023/03/24/store-editing-roadmap-update-q1-2023/
and it doesn't really have a lot of details.

Would you say looking at github milestones is the best approach? 
Like:
https://github.com/woocommerce/woocommerce-blocks/milestone/241

**Comment by senadir** [comment]
> I think it's great that your team is asking for feedback but I have a hard time finding something official on what's being worked on etc.

We're moving toward to having public communication, you can follow [this roadmap](https://developer.woo.com/roadmap/) which includes all work in WooCommerce, not just Checkout.

We will also update it early 2024 for our plans for it. Right now, our priority is getting more extensibility options out. We identified adding custom fields as a top priority.

> May I ask if there's any plans to officially support this approach and possibly add that capability without a CSS based solution?

We don't have plans for that yet as far as I know.

**Comment by pierorocca** [comment]
Closing this issue as checkout is designed to be optimized for use with the order summary, a standard checkout pattern.
